tests/v2/validation/steve/vai/vai.go

```diff
@@ -0,0 +1,62 @@
+package vai
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/extensions/vai"
+)
+
+const randomStringLength = 8
+
+type SupportedWithVai interface {
+	SupportedWithVai() bool
+}
+
+func enableVAI(client *rancher.Client, vaiEnabled *bool, once *sync.Once) error {
+	var enableErr error
+	once.Do(func() {
+		if err := vai.EnableVaiCaching(client); err != nil {
+			enableErr = fmt.Errorf("failed to enable VAI caching: %w", err)
+			return
+		}
+		*vaiEnabled = true
+		time.Sleep(time.Second * 60)
```

Comment:
I couldn't get a reliable polling mech to work decently here. It's not about the pods being ready. (Rancher pods get restarted) but just because the pods are successfully running and restarted doesn't mean the tests can run, because it's waiting on the SQL database files within the rancher pods

Reply:
@brudnak what happens if test starts too early?

Expected behavior is that API calls just block until the SQL database files are ready. As soon as they start responding, responses should be correct.

Have you observed anything different?

Reply:
seeing that described behavior now. what I noticed ~1month ago doesn't seem present now.